db/compaction/compaction_job.cc

```diff
@@ -245,7 +245,7 @@ void CompactionJob::ReportStartedCompaction(Compaction* compaction) {
   compaction_job_stats_->is_full_compaction = compaction->is_full_compaction();
 }
 
-void CompactionJob::Prepare() {
+void CompactionJob::PrepareSubs() {
```

Comment:
nit: we weren't doing this before, but since we are making changes here now. Do we want to add a quick assertion to make sure mutex is held? And same for `PrepareTimes()`

```
db_mutex_->AssertHeld();
```

Reply:
Thanks for the call-out as I was violating my own (somewhat copied) contract and that could have been a race.